go/cmd/dolt/commands/log.go

```diff
@@ -198,3 +211,132 @@ func logCommits(ctx context.Context, dEnv *env.DoltEnv, cs *doltdb.CommitSpec, l
 
 	return 0
 }
+
+func checkIfTableExists(ctx context.Context, commit *doltdb.Commit, tableName string) (bool, error) {
+	rv, err := commit.GetRootValue()
+	if err != nil {
+		return false, err
+	}
+
+	_, ok, err := rv.GetTable(ctx, tableName)
+	if err != nil {
+		return false, err
+	}
+
+	return ok, nil
+}
+
+func logTableCommits(ctx context.Context, dEnv *env.DoltEnv, loggerFunc commitLoggerFunc, tableName string, numLines int) int {
+	commit, err := dEnv.DoltDB.Resolve(ctx, dEnv.RepoState.CWBHeadSpec(), dEnv.RepoState.CWBHeadRef())
+	if err != nil {
+		cli.PrintErrln(color.HiRedString("Fatal error: cannot get HEAD commit for current branch."))
+		return 1
+	}
+
+	// Check that the table exists in the head commit
+	exists, err := checkIfTableExists(ctx, commit, tableName)
+	if err != nil {
+		cli.PrintErrln(color.HiRedString("Fatal error: failed to get commit info"))
+		return 1
+	}
+
+	if !exists {
+		cli.PrintErrln("error: Invalid commit spec or table given")
+		return 1
+	}
+
+	h, err := commit.HashOf()
+	if err != nil {
+		cli.PrintErrln(color.HiRedString("Fatal error: failed to get commit hash"))
+		return 1
+	}
+
+	commits, err := commitwalk.GetTopNTopoOrderedCommits(ctx, dEnv.DoltDB, h, -1)
```

Comment:
Don't do this, this fetches the entire commit history one commit at a time, you don't want that. Use `GetTopologicalOrderIterator` and iterate